node/src/components/block_accumulator.rs

```diff
@@ -616,62 +671,26 @@ impl<REv: ReactorEvent> Component<REv> for BlockAccumulator {
 
 impl<REv: ReactorEvent> ValidatorBoundComponent<REv> for BlockAccumulator {
     fn handle_validators(&mut self, effect_builder: EffectBuilder<REv>) -> Effects<Self::Event> {
-        let mut effects = Effects::new();
-        for block_acceptor in self
+        let validator_matrix = &self.validator_matrix; // Closure can't borrow all of self.
+        let should_stores = self
             .block_acceptors
             .values_mut()
             .filter(|acceptor| false == acceptor.has_sufficient_finality())
-        {
-            if let Some(era_id) = block_acceptor.era_id() {
-                if let Some(evw) = self.validator_matrix.validator_weights(era_id) {
-                    effects.extend(store_block_and_finality_signatures(
-                        effect_builder,
-                        block_acceptor.should_store_block(&evw),
-                    ));
-                }
-            }
-        }
-        effects
+            .filter_map(|acceptor| {
+                let era_id = acceptor.era_id()?;
+                let evw = validator_matrix.validator_weights(era_id)?;
+                Some(acceptor.should_store_block(&evw))
+            })
+            .collect_vec();
```

Comment:
nit part 3:
```suggestion
            .collect();
```

Reply:
I tried it, but then Clippy complains due to https://github.com/rust-lang/rust-clippy/issues/9219.
Since we're already using itertools elsewhere anyway, I'd postpone that change until the Clippy bug is fixed.